### PR TITLE
Fixes #329  : menu closes if open when user clicks or taps away

### LINKED ISSUE #329: Menu: Close menu if open when user clicks or taps away
In the new mobile nav component, when a click or tap event happens outside of the mobile nav while a menu is open, all menus should close.

### PR DESCRIPTION
**Pull request by anishagg17**
**Description**

menu closes if open when user clicks or taps away

**Technical**

A new component is added which detects click outside it

**Evidence**

Hnet com-image (3)

### CODE REVIEW

**Comment by anishagg17** [comment]
@shaneriley i know you are busy but please can you review this pr?

File: packages/ia-components/sandbox/language-select/CloseMenu.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React, { Component } from "react";
+
+/**
+ * Component that alerts if you click outside of it
+ */
+export default class CloseMenu extends Component {
+  constructor(props) {
+    super(props);
+
+    this.setWrapperRef = this.setWrapperRef.bind(this);
```
Comment (shaneriley) [inline packages/ia-components/sandbox/language-select/CloseMenu.js]: Please bind these methods using `forEach` as we do in other component constructors like https://github.com/internetarchive/iaux/blob/master/packages/ia-components/sandbox/language-select/language-select.js#L80

File: packages/ia-components/sandbox/language-select/CloseMenu.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener("mousedown", this.handleClickOutside);
```
Comment (shaneriley) [inline packages/ia-components/sandbox/language-select/CloseMenu.js]: Please change all quotation marks to single quotes.

File: packages/ia-components/sandbox/language-select/language-select.js [hunk 3]
```diff
@@ -102,37 +101,50 @@ class LanguageSelect extends React.Component {
...
-          {Object.keys(this.languages).map((language) => (
-            <LanguageOption
-               key={`language_${language}`}
-               language={this.languages[language]}
-               value={language}
-               selectedLanguage={selectedLanguage}
-               onClick={this.selectLanguage} />))}
-        </ul>
+        {dropdownOpen && (
+          <ul className={dropdownOpen ? "visible" : null}>
```
Comment (shaneriley) [inline packages/ia-components/sandbox/language-select/language-select.js]: Now that a short-circuit is used to conditionally render this entire block, the ternary for setting a class on this element is redundant. Remove and set the `className` to always be 'visible'.

**Comment by shaneriley** [comment]
@anishagg17 The issue this PR references is for the mobile nav component, not for the language select. Unfortunately, the description for the issue was never filled in, so it wasn't clear that it was for a particular component. While this is a welcome addition to the language select, it does not satisfy the criteria for closing issue #329 . Please amend the PR description to remove the 'Closes' line.

**Comment by anishagg17** [comment]
Done ! @shaneriley

**Comment by anishagg17** [comment]
Done ! @shaneriley

**Comment by anishagg17** [comment]
there is no .prettierrc file in this repo that's why my local prettier settings turned those single quotes into double, @shaneriley do you think that a .prettierrc file should be included into the root of this repository

**Pull request closed without merging**